Lib/lib2to3/tests/test_parser.py

```diff
@@ -212,12 +215,33 @@ def test_multiline_str_literals(self):
         self.validate(s)
 
 
+
 def diff(fn, result, encoding):
-    "A diff the result and original file content independent of OS."
+    """Diff the result and original file content independent of OS. This 
+    implementation is in Jython only. CPython 2.x calls out to the shell diff
+    command. CPython 3.x str equality behaviour is simpler and should allow
+    this test file to be replaced wholesale in Jython 3.x."""
     r = iter(result.encode(encoding).splitlines(True))
-    with open(fn, "rb") as f:
```

Comment:
Ah, ok. Here's the nub of it. r is a list of unicode but the file is being read as binary?

Reply:
In general it can be unicode, but actually the file it fails on just has ascii and CRLF.

Reply:
Oh hang on, looking properly and after coffee ... . This is an **encode**. So `result` should be `unicode` (or `bytes` to be interpreted as ASCII characters in Python 2) and `r` is definitely a sequence of `bytes` objects. So wasn't "rb" the right thing to do? Unless of course the encoding of what's in the file differs from `encoding`.

I notice in the CPython original that this test doesn't try to work on Windows, so it may not have dealt correctly with CRLF. If CRLF in the file is the problem, isn't the answer "rbU"?

Reply:
That makes sense. I actually didn't know about file mode 'U'. In latest patch. Understood this will probably sit for a bit while the beta progresses.